habitat-lab/habitat/tasks/rearrange/actions/actions.py

```diff
@@ -469,6 +469,163 @@ def step(self, *args, is_last_action, **kwargs):
+
+    Optionally cull states with active collisions if config parameter `allow_dyn_slide` is True
+    """
+
+    def __init__(self, *args, config, sim: RearrangeSim, **kwargs):
+        super().__init__(*args, config=config, sim=sim, **kwargs)
+        self._sim: RearrangeSim = sim
+        self.base_vel_ctrl = habitat_sim.physics.VelocityControl()
+        self.base_vel_ctrl.controlling_lin_vel = True
+        self.base_vel_ctrl.lin_vel_is_local = True
+        self.base_vel_ctrl.controlling_ang_vel = True
+        self.base_vel_ctrl.ang_vel_is_local = True
+        self._allow_dyn_slide = self._config.get("allow_dyn_slide", True)
+        self._lin_speed = self._config.lin_speed
+        self._ang_speed = self._config.ang_speed
+        self._allow_back = self._config.allow_back
+        self._cylinder_deviate = self._config.cylinder_deviate
+        self._lin_collision_threshold = self._config.lin_collision_threshold
+        self._ang_collision_threshold = self._config.ang_collision_threshold
+
+    @property
+    def action_space(self):
+        lim = 20
+        return spaces.Dict(
+            {
+                self._action_arg_prefix
+                + "base_vel": spaces.Box(
+                    shape=(2,), low=-lim, high=lim, dtype=np.float32
+                )
+            }
+        )
+
+    def _capture_articulated_agent_state(self):
+        return {
+            "forces": self.cur_articulated_agent.sim_obj.joint_forces,
+            "vel": self.cur_articulated_agent.sim_obj.joint_velocities,
+            "pos": self.cur_articulated_agent.sim_obj.joint_positions,
+        }
+
+    def _set_articulated_agent_state(self, set_dat):
+        self.cur_articulated_agent.sim_obj.joint_positions = set_dat["forces"]
+        self.cur_articulated_agent.sim_obj.joint_velocities = set_dat["vel"]
+        self.cur_articulated_agent.sim_obj.joint_forces = set_dat["pos"]
+
+    def spot_collison_check(self, trans, target_trans, end_pos):
+        # Get the front and rear positions
+        front_cur_pos = trans.transform_point(
+            np.array([self._cylinder_deviate, 0, 0])
+        )
+        front_goal_pos = target_trans.transform_point(
+            np.array([self._cylinder_deviate, 0, 0])
+        )
+        rear_cur_pos = trans.transform_point(
+            np.array([-self._cylinder_deviate, 0, 0])
+        )
+        rear_goal_pos = target_trans.transform_point(
+            np.array([-self._cylinder_deviate, 0, 0])
+        )
+
+        # For step filter of the front pos
+        end_front_pos = self._sim.step_filter(front_cur_pos, front_goal_pos)
+        end_front_pos[1] = front_cur_pos[1]
+        front_goal_pos[1] = front_cur_pos[1]
+        # For step filter of the rear pos
+        end_rear_pos = self._sim.step_filter(rear_cur_pos, rear_goal_pos)
+        end_rear_pos[1] = front_cur_pos[1]
+        rear_goal_pos[1] = front_cur_pos[1]
```

Comment:
I'm assuming you want to project all of these points to the XZ plane so distances are not polluted by changes in height.
```suggestion
        end_rear_pos[1] = rear_cur_pos[1]
        rear_goal_pos[1] = rear_cur_pos[1]
```

Reply:
Good catch! Thanks for this!